### DISCUSSION TITLE
Setting up rules to control pwm channel

### DISCUSSION THREAD

**Discussion by noroozihossein** [arendst/Tasmota#15386] [Q&A] [answered]
Hello guys , anybody could help,I will be appreciated , i have a ESP32 module tasmota installed, setting up 3 channel pwm Dimmer for lighting control and also have 3 push button related to other gpio of ESP32 to control each of pwm channel, what should i setup via rules to do short press on/off each pwm channel, long press increase/decrease lighting lux level.

**Comment by noroozihossein** [comment, marked as answer]
> Hello guys , anybody could help,I will be appreciated , i have a ESP32 module tasmota installed, setting up 3 channel pwm Dimmer for lighting control and also have 3 push button related to other gpio of ESP32 to control each of pwm channel, what should i setup via rules to do short press on/off each pwm channel, long press increase/decrease lighting lux level.

I know that the switchmode should be setup to switchmode 11 , but don’t know after that what rules should setup to do exactly same scenario

**Reply by barbudor** [reply]
Here you will find examples of rules that work with SwitchMode 11 to control a Dimmer with a single push-button:
https://tasmota.github.io/docs/Rules/#control-a-dimmer-with-one-switch

This is already linked at the end of the SwitchMode 11 description in "Buttons & Switches"
